pkg/persistentcache/persistentcache.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019 Datadog, Inc.
+
+package persistentcache
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// Return a file where to store the data. We split the key by ":", using the
+// first prefix as directory, if present. This is useful for integrations, which
+// use the check_id formed with $check_name:$hash
+func getFileForKey(key string) (string, error) {
+	// Invalid characters to clean up
+	invalidChars, err := regexp.Compile("[<>:\"\\|?/\\*]")
```

Comment:
This won't work on windows where the folder separator is `\`

```golang
cleanedPath := invalidChars.ReplaceAllString("\\..\\..\\Program Files\\file", "")
# cleanedPath: "..\\..\\Program Files\file"
}
```

Reply:
You're right, used a whitelist instead.